src/components/PhoneInput/PhoneInput.jsx

```diff
@@ -0,0 +1,134 @@
+// @flow
+import React, {
+  Component,
+  type Node,
+  type ComponentType,
+} from 'react';
+import { PhoneInput as ReactPhoneInput } from 'react-phone-number-input';
+import { SmartInput } from 'react-phone-number-input/smart-input';
+import { getExampleNumber } from 'libphonenumber-js';
+
+import metadata from 'libphonenumber-js/metadata.min.json';
+import examples from 'libphonenumber-js/examples.mobile.json';
+import labels from 'react-phone-number-input/locale/default.json';
+
+import { getDataAttrs } from '../../dataUtils';
+import type { Data } from '../../types';
+
+import CountrySelect from './CountrySelect';
+import FlagIcon from './FlagIcon';
+
+import type {
+  Flags,
+  Labels,
+  Metadata,
+  PhoneNumber,
+} from './types';
+
+import styles from './style.scss';
+
+export type InputComponentProps = {|
+  value: string,
+  onChange?: (value: ?string) => void,
+  onFocus?: () => void,
+  onBlur?: () => void,
+  country?: ?string,
+  metadata: Metadata,
+|};
+
+type State = {|
+  country: ?string,
+|};
+
+export type Props = {|
+  value?: ?string,
+  onChange?: (value: ?string) => void,
+  onFocus?: <+T: EventTarget = EventTarget>(SyntheticFocusEvent<T>) => void,
+  onBlur?: <+T: EventTarget = EventTarget>(SyntheticFocusEvent<T>) => void,
+  onKeyDown?: <+T: EventTarget = EventTarget>(SyntheticKeyboardEvent<T>) => void,
+  disabled: boolean,
+  autoComplete: string,
+  displayInitialValueAsLocalNumber: boolean,
+  country?: string,
+  countries?: Array<string>,
+  labels: Labels,
+  flagsPath: string,
+  flags: Flags,
+  flagComponent: ComponentType<$Shape<Props>>,
+  international: boolean,
+  internationalIcon: () => ComponentType<*>,
+  showCountrySelect: boolean,
+  countrySelectTabIndex?: number,
+  countryOptions?: Array<string>,
+  style?: { [key: string]: string },
+  className?: string,
+  inputClassName?: string,
+  getInputClassName?: ({| disabled?: boolean, invalid?: boolean |}) => string,
+  countrySelectComponent: ComponentType<$Shape<Props>>,
+  inputComponent: ComponentType<$Shape<InputComponentProps>>,
+  ext?: Node,
+  limitMaxLength: boolean,
+  error?: string,
+  indicateInvalid?: boolean,
+  locale?: { [key: string]: string },
+  metadata: Metadata,
+  onCountryChange: (country: ?string) => void,
+  placeholder?: string,
+  data: Data,
+  [prop: string]: any,
```

Comment:
why add this wildcard?

Reply:
I can't remember my exact motivation at the moment, but there are keys that do not appear here that should be possible to add on this component.
For example: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes